# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: js/src/wasm/WasmConstants.h
# Commit: 00d2290648a7
# Full Hash: 00d2290648a75aeefbb4738298ddfcda2d0797b0
# Author: Ryan Hunt <rhunt@eqrion.net>
# Date: 2020-08-18 09:24:52
# Regressor Bug: 1561521
# File Overlap Count: 2
# Description:
#   Bug 1561521 - Add non-nullable references. r=lth
#   
#   This commit adds decoding, validation, and JS-API type-checking for
#   non-nullable references.
#   
# ==============================================================================

diff -r cd847eba7f9c -r 00d2290648a7 js/src/wasm/WasmConstants.h
--- a/js/src/wasm/WasmConstants.h	Tue Aug 18 00:06:00 2020 +0000
+++ b/js/src/wasm/WasmConstants.h	Tue Aug 18 00:40:20 2020 +0000
@@ -65,8 +65,11 @@
   // A reference to any host value.
   ExternRef = 0x6f,  // SLEB128(-0x11)
 
-  // Type constructor for reference types.
-  OptRef = 0x6c,
+  // Type constructor for nullable reference types.
+  NullableRef = 0x6c,  // SLEB128(-0x14)
+
+  // Type constructor for non-nullable reference types.
+  Ref = 0x6b,  // SLEB128(-0x15)
 
   // Type constructor for function types
   Func = 0x60,  // SLEB128(-0x20)
@@ -91,6 +94,11 @@
 
 static constexpr TypeCode AbstractReferenceTypeCode = TypeCode::ExternRef;
 
+// A type code used to represent (ref null? typeindex) whether or not the type
+// is encoded with 'Ref' or 'NullableRef'.
+
+static constexpr TypeCode AbstractReferenceTypeIndexCode = TypeCode::Ref;
+
 enum class FuncTypeIdDescKind { None, Immediate, Global };
 
 // A wasm::Trap represents a wasm-defined trap that can occur during execution